File: src_original/extras-kio/libksslerroruidata.cpp
#include <KSslErrorUiData>
#include <QList>
#include <QNetworkReply>
#include <QSslError>
#include <QSslSocket>
#include <ksslerroruidata.h>
#include "libksslerroruidata.h"
#include "libksslerroruidata.hxx"

KSslErrorUiData* KSslErrorUiData_new() {
    return new KSslErrorUiData();
}

KSslErrorUiData* KSslErrorUiData_new2(const QSslSocket* socket) {
    return new KSslErrorUiData(socket);
}

KSslErrorUiData* KSslErrorUiData_new3(const QNetworkReply* reply, const libqt_list /* of QSslError* */ sslErrors) {
    QList<QSslError> sslErrors_QList;
    sslErrors_QList.reserve(sslErrors.len);
    QSslError** sslErrors_arr = static_cast<QSslError**>(sslErrors.data);
    for (size_t i = 0; i < sslErrors.len; ++i) {
        sslErrors_QList.push_back(*(sslErrors_arr[i]));
    }
    return new KSslErrorUiData(reply, sslErrors_QList);
}

KSslErrorUiData* KSslErrorUiData_new4(const KSslErrorUiData* other) {
    return new KSslErrorUiData(*other);
}

void KSslErrorUiData_OperatorAssign(KSslErrorUiData* self, const KSslErrorUiData* param1) {
    self->operator=(*param1);
}

void KSslErrorUiData_Delete(KSslErrorUiData* self) {
    delete self;
}

File: include_original/extras-kio/libksslerroruidata.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const std = @import("std");

/// https://api.kde.org/ksslerroruidata.html
pub const ksslerroruidata = struct {
    /// New constructs a new KSslErrorUiData object.
    ///
    ///
    pub fn New() QtC.KSslErrorUiData {
        return qtc.KSslErrorUiData_new();
    }

    /// New2 constructs a new KSslErrorUiData object.
    ///
    /// ``` socket: QtC.QSslSocket ```
    pub fn New2(socket: ?*anyopaque) QtC.KSslErrorUiData {
        return qtc.KSslErrorUiData_new2(@ptrCast(socket));
    }

    /// New3 constructs a new KSslErrorUiData object.
    ///
    /// ``` reply: QtC.QNetworkReply, sslErrors: []QtC.QSslError ```
    pub fn New3(reply: ?*anyopaque, sslErrors: []QtC.QSslError) QtC.KSslErrorUiData {
        const sslErrors_list = qtc.libqt_list{
            .len = sslErrors.len,
            .data = @ptrCast(sslErrors.ptr),
        };

        return qtc.KSslErrorUiData_new3(@ptrCast(reply), sslErrors_list);
    }

    /// New4 constructs a new KSslErrorUiData object.
    ///
    /// ``` other: QtC.KSslErrorUiData ```
    pub fn New4(other: ?*anyopaque) QtC.KSslErrorUiData {
        return qtc.KSslErrorUiData_new4(@ptrCast(other));
    }

    /// [Qt documentation](https://api.kde.org/ksslerroruidata.html#operator-eq)
    ///
    /// ``` self: QtC.KSslErrorUiData, param1: QtC.KSslErrorUiData ```
    pub fn OperatorAssign(self: ?*anyopaque, param1: ?*anyopaque) void {
        qtc.KSslErrorUiData_OperatorAssign(@ptrCast(self), @ptrCast(param1));
    }

    /// [Qt documentation](https://api.kde.org/ksslerroruidata.html#dtor.KSslErrorUiData)
    ///
    /// Delete this object from C++ memory.
    ///
    /// ``` self: QtC.KSslErrorUiData ```
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.KSslErrorUiData_Delete(@ptrCast(self));
    }
};
